api/actions/listing.py

```diff
@@ -93,6 +93,53 @@ async def get_relevant_addons(user: UserEntity) -> tuple[str, list[BaseServerAdd
     )
 
 
+def match_list_subtypes(
+    action_subtypes: list[str],
+    context_subtypes: list[str],
+) -> bool:
+    """List matching
+
+    ## action subtype (defined by manifest)
+
+    - always includes the contained entity type (version, folder, etc.)
+    - optionally includes the list type: "version:review-session"
+
+    ## context subtypes (provided by the client)
+
+    always includes both parts: "version:review-session"
+
+    ## Matching Rules
+
+    An action_subtype matches a context_subtype if:
+
+    - The contained type matches (version == version)
+
+    - And if action_subtype includes a list type,
+      it must also match (review-session)
+
+    - If action_subtype doesn't include a list type,
+      it's considered a wildcard (matches all list types
+      of that entity type)
+    """
+    for context_subtype in context_subtypes:
```

Comment:
Shouldn't this be more like this?
```suggestion
    if not action_subtypes or None in action_subtypes:
        return True

    for context_subtype in context_subtypes:
```